x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -190,11 +190,41 @@ export function trainedModelsRoutes(
             // we don't need to fill kibana's log with these messages.
             mlLog.debug(e);
           }
+          const enabledFeatures = getEnabledFeatures();
+          try {
+            if (enabledFeatures.dfa) {
+              const jobIds = result.map((model) => {
+                let id = model.metadata?.analytics_config?.id;
+                if (id) {
+                  id = `${id}*`;
+                }
+                return id;
+              });
+              const filteredJobIds = jobIds.filter((id) => id !== undefined);
 
-          const body = filterForEnabledFeatureModels(result, getEnabledFeatures());
+              const { data_frame_analytics: jobs } = await mlClient.getDataFrameAnalytics({
+                id: filteredJobIds.join(','),
+                allow_no_match: true,
+              });
+
+              jobs.forEach(({ id }) => {
+                const model = result.find(
```

Comment:
This would only set it to true if a job id returned from the check matched the job id on the model. But agree that we should be setting it explicitly to false instead of just not adding the property at all and relying on falsey-ness.

Reply:
Updated to set explicitly in 50ffcfdd594f6cedb13acedc195b6cd00bac950d

Reply:
Exactly yeah,
`origin_job_exists === undefined` -> this is not a DFA model
`origin_job_exists === true` -> this a DFA model and the job exists
`origin_job_exists === false` -> this is a DFA model and the job no longer exits.
Falsey-ness is evil :D